lib/rules/no-smart-quotes.js

```diff
@@ -41,29 +49,47 @@ module.exports = {
                     return;
                 }
 
-                const match = node.raw.match(/^(.*(“|”|‘|’).*)$/);
-
-                if (match) {
-                    const offendingPunctuation = match[2];
-                    const matchedString = match[1];
-                    const alternate = {
-                        '“': '"',
-                        '”': '"',
-                        '‘': "'",
-                        '’': "'",
-                    } [offendingPunctuation];
-
-                    if (typeof matchedString !== 'undefined' && typeof offendingPunctuation !== 'undefined') {
-                        const message = `Strings must use straight quotes in: ${ matchedString}. Use ${alternate} instead.`;
-                        context.report({
-                            node,
-                            message: message,
-                            data: {
-                                offendingPunctuation
-                            }
-                        });
-                    }
+                const matches = Array.from(matchAll(node.raw, SMART_QUOTES_REGEX));
+
+                if (matches.length === 0) {
+                    return;
                 }
+
+                context.report({
+                    node,
+                    message: "Strings must use straight quotes.",
+                    fix(fixer) {
+                        const stringQuoteType = node.raw.charAt(0);
+
+                        let fixed = ""
+                        let matchIndex = 0;
+
+                        for (let index = 0; index < node.raw.length; index++) {
```

Comment:
Note: This loop is a little bit complicated in order to make sure that replacing `“` with `\"` doesn't screw up the current string index.